core/src/main/resources/error/README.md

```diff
@@ -857,6 +857,17 @@ The following SQLSTATEs are collated from:
 |42K0B    |42   |Syntax error or Access Rule violation             |K0B     |Legacy feature blocked                                      |Spark          |N       |Spark                                                                       |
+|42KD0    |42   |Syntax error or Access Rule violation             |KD0     |Ambiguous name reference.                                   |Databricks     |N       |Databricks                                                                  |
+|42KD1    |42   |Syntax error or Access Rule violation             |KD1     |Operation not supported in READ ONLY session mode.          |Databricks     |N       |Databricks                                                                  |
+|42KD2    |42   |Syntax error or Access Rule violation             |KD2     |The source and target table names of a SYNC operaton must be the same.|Databricks     |N       |Databricks                                                                  |
+|42KD3    |42   |Syntax error or Access Rule violation             |KD3     |A column can not be added as specified.                     |Databricks     |N       |Databricks                                                                  |
+|42KD4    |42   |Syntax error or Access Rule violation             |KD4     |Operation not supported because table schema has changed.   |Databricks     |N       |Databricks                                                                  |
+|42KD5    |42   |Syntax error or Access Rule violation             |KD5     |Cannot create file or path.                                 |Databricks     |N       |Databricks                                                                  |
+|42KD6    |42   |Syntax error or Access Rule violation             |KD6     |No partition information found.                             |Databricks     |N       |Databricks                                                                  |
+|42KD7    |42   |Syntax error or Access Rule violation             |KD7     |Table signature mismatch.                                   |Databricks     |N       |Databricks                                                                  |
+|42KD8    |42   |Syntax error or Access Rule violation             |KD8     |Column position out of range.                               |Databricks     |N       |Databricks                                                                  |
+|42KD9    |42   |Syntax error or Access Rule violation             |KD9     |Cannot infer table schema.                                  |Databricks     |N       |Databricks                                                                  |
+|42KDA    |42   |Syntax error or Access Rule violation             |KDA     |Failed to merge file into table schema.                     |Databricks     |N       |Databricks                                                                  |
```

Comment:
cc @srielau do we want to add those SQLSTATE to `core/src/main/resources/error/README.md` ?

Otherwise, we can't use them in error class because we only allow the SQLSTATE specified in `core/src/main/resources/error/README.md`.

For example, we cannot apply https://github.com/apache/spark/pull/39871#discussion_r1095930407.

Reply:
Yes

Reply:
Also cc @MaxGekk @cloud-fan